drwmutex.go

```diff
@@ -89,15 +96,22 @@ func (dm *DRWMutex) Lock() {
 func (dm *DRWMutex) RLock() {
 
 	isReadLock := true
-	dm.lockBlocking(isReadLock)
+	dm.lockBlocking(time.Duration(1<<63-1), isReadLock)
+}
+
+// GetRLock tries to get a read lock on dm before the timeout occurs.
+func (dm *DRWMutex) GetRLock(timeout time.Duration) (locked bool) {
+
+	isReadLock := true
+	return dm.lockBlocking(timeout, isReadLock)
 }
 
 // lockBlocking will acquire either a read or a write lock
 //
 // The call will block until the lock is granted using a built-in
 // timing randomized back-off algorithm to try again until successful
-func (dm *DRWMutex) lockBlocking(isReadLock bool) {
-	doneCh := make(chan struct{})
+func (dm *DRWMutex) lockBlocking(timeout time.Duration, isReadLock bool) bool {
+	doneCh, start := make(chan struct{}), time.Now()
```

Comment:
Given the very limited scope (just in relation to the time.Since a couple of lines down below), and as we are not storing it, time zones issues shouldn't come into play. So it would be a bit overkill here.